File: packages/core/src/types/tenant.ts
export type TenantPlan = "trial" | "review" | "loyalty" | "pro" | "network";

export type BusinessType =
  | "barbershop"
  | "cafe"
  | "pet_grooming"
  | "salon"
  | "tattoo"
  | "other";

export interface Tenant {
  id: string;
  slug: string;
  name: string;
  business_type: BusinessType;
  logo_url: string | null;
  primary_color: string;
  accent_color: string;
  google_place_id: string | null;
  google_review_url: string | null;
  google_business_url: string | null;
  stamps_for_reward: number;
  reward_description: string | null;
  reward_expires_days: number;
  stamp_rate_limit_minutes: number;
  plan: TenantPlan;
  is_founding_member: boolean;
  stripe_customer_id: string | null;
  stripe_subscription_id: string | null;
  trial_ends_at: string | null;
  is_active: boolean;
  cancelled_at: string | null;
  created_at: string;
  updated_at: string;
}

File: backend/app/schemas/tenant.py
from typing import Literal

from pydantic import BaseModel, Field

HexColor = str


class TenantSettingsUpdateIn(BaseModel):
    name: str | None = Field(default=None, min_length=2, max_length=80)
    logo_url: str | None = None
    primary_color: HexColor | None = Field(default=None, pattern=r"^#[0-9A-Fa-f]{6}$")
    accent_color: HexColor | None = Field(default=None, pattern=r"^#[0-9A-Fa-f]{6}$")
    google_place_id: str | None = None
    google_review_url: str | None = None
    google_business_url: str | None = None


class RewardConfigIn(BaseModel):
    stamps_for_reward: int = Field(ge=1, le=50)
    reward_description: str = Field(min_length=2, max_length=120)
    reward_expires_days: int = Field(ge=1, le=365)
    stamp_rate_limit_minutes: int = Field(ge=0, le=1440)


BusinessType = Literal["barbershop", "cafe", "pet_grooming", "salon", "tattoo", "other"]
